src/main/java/ru/itis/controllers/WishListController.java

```diff
@@ -1,56 +1,56 @@
-//package ru.itis.controllers;
-//
-//import io.swagger.annotations.ApiOperation;
-//import org.springframework.beans.factory.annotation.Autowired;
-//import org.springframework.http.ResponseEntity;
-//import org.springframework.security.access.prepost.PreAuthorize;
-//import org.springframework.stereotype.Controller;
-//import org.springframework.web.bind.annotation.*;
-//import ru.itis.dto.WishListDto;
-//import ru.itis.forms.ItemForm;
-//import ru.itis.models.WishList;
-//import ru.itis.services.ItemService;
-//import ru.itis.services.WishListService;
-//
-//import java.util.Optional;
-//
-//import static ru.itis.dto.WishListDto.from;
-//
-//@Controller
-//public class WishListController {
-//
-//    @Autowired
-//    private ItemService itemService;
-//
-//    @Autowired
-//    private WishListService wishListService;
-//
-//    @CrossOrigin
-//    @GetMapping("/lists/{list-id}")
-//    @PreAuthorize("isAuthenticated()")
-//    @ApiOperation("Get a single wish list")
-//    public ResponseEntity<Object> getWishListPage(@PathVariable(name = "list-id") Long listId) {
-//        Optional<WishList> wishListCandidate = wishListService.findWishListById(listId);
-//        if (wishListCandidate.isPresent()) {
-//            WishListDto wishListDto = from(wishListCandidate.get());
-//            return ResponseEntity.ok(wishListDto);
-//        } else throw new IllegalArgumentException("Can not find such wish list");
-//    }
-//
-//    @CrossOrigin
-//    @PostMapping("/lists/{list-id}")
-//    @PreAuthorize("isAuthenticated()")
-//    @ApiOperation("Add new item to wish list")
-//    public ResponseEntity<Object> addNewItem(@PathVariable(name = "list-id") Long listId, ItemForm itemForm) {
-//        return ResponseEntity.ok(itemService.addNewItem(itemForm));
-//    }
-//
-//    @CrossOrigin
-//    @PostMapping("/lists/{list-id}")
-//    @PreAuthorize("isAuthenticated()")
-//    @ApiOperation("Delete an item")
-//    public ResponseEntity<Object> deleteItem(@PathVariable(name = "list-id") Long listId ,@RequestParam(name = "title") String itemName) {
-//        itemService.removeByName(itemName);
-//        return ResponseEntity.ok().build();
-//    }
-//}
+package ru.itis.controllers;
+
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import ru.itis.dto.WishListDto;
+import ru.itis.forms.ItemForm;
+import ru.itis.models.WishList;
+import ru.itis.services.ItemService;
+import ru.itis.services.WishListService;
+
+import java.util.Optional;
+
+import static ru.itis.dto.WishListDto.from;
+
+@Controller
+public class WishListController {
+
+    @Autowired
+    public ItemService itemService;
+
+    @Autowired
+    public WishListService wishListService;
```

Comment:
Why? Using @Autowired on constructors or setters even worse

Reply:
Top `@Autowired` usage:
1. Constructors. <- ok
2. Setters. <- bad
3. Attributes. <- worst case ever

Setters are okaaaaay, but you can miss some field while creating your object. That will cause NullPointerException at runtime (waste of time). Constructors are good, because you always know what your object need to be created. You can't miss anything.
Attributes are just trash, how would you test your code? How to create such an object?
```
@Before void constructMe() {
    Service s = new Service();
    s.getClass().getAttribute("dao").set(s, new Dao());
    s.getClass().getAttribute("api").set(s, new Api());
}
```

Reply:
see https://dzone.com/articles/spring-di-patterns-the-good-the-bad-and-the-ugly

Reply:
fixed